src/internal/internetReachability.ts

```diff
@@ -41,14 +38,14 @@ function checkInternetReachability(): InternetReachabilityCheckHandler {
   // We wraop the promise to allow us to cancel the pending request, if needed
   let hasCanceled = false;
 
-  const promise = fetch(REACHABILITY_URL)
+  const promise = fetch(netInfoConfig.reachabilityUrl)
     .then(
       (response): void => {
         if (!hasCanceled) {
           setIsInternetReachable(response.status === 204);
```

Comment:
In the configuration, as well as being able to configure the URL, we also need to be able to supply a function which checks if the internet is reachable. This is because each URL will have a different expected way of saying success.

I have added some suggestions around this review so you can see what I mean. All of them are untested, so you might need to make some changes to get them to work.